### ISSUE TITLE
Fullscreen mode can cause kernel panic on OSX Sonoma

### ISSUE THREAD

**Issue by Rydier** [guillep/pharo-core#16451]
I don't know if it's just my machine, but for me, I'm consistently getting a kernel panic that restarts the machine, shortly after switching Pharo to fullscreen mode. 

Freshly downloaded Pharo 12 image, launched from the PharoLaunceher, and using the 2024-03-12 VM running on an Macbook M2 Pro with Sonoma 14.4.1.

The crash report window lists the cause as being
program_swap: Async Swap request landing on unsupported platform. Force panic.

It happens both when running on the built-in display, and when Pharo goes fullscreen on an external monitor.
When I tried it on the built-in display, the panic occurred right away.
On an external monitor, I was able to open a browser, before the bottom half of the screen started going green, then the panic occurs shortly after.

It appears similar to issues reported for other software when running on Sonoma:
https://community.native-instruments.com/discussion/27721/kontakt-7-window-across-multiple-screens-causing-kernel-panic-shutoff-on-mac-studio-kt-8785/p5

**Comment by Ducasse** [comment]
Thanks for the report. I will try because I'm also on Sonora 14.1.
@tesonep

**Comment by jordanmontt** [comment]
I don't know if related, but I also experienced machine crashes (the whole machine restarts) while using Pharo in Sonoma 14.4.1. I used the latest VM and image. It happened me 3 times in one day

**Comment by svenvc** [comment]
me too, once, can't repeat it

**Comment by jordanmontt** [comment]
It happened today again several times. For information, I was not using fullscreen mode in any of the times when the machine crashed.

**Comment by jbrichau** [comment]
I have such crashes daily since updating from Ventura to Sonoma but it's not specific to Pharo. I had crashes while not using Pharo at all with the same error report (e.g. while exporting a pdf from Preview).
Just wanted to register that here, that in my experience, it might not be related to Pharo but entirely to Mac Os Sonoma.

**Comment by jordanmontt** [comment]
Yes, After some days I realized that too... It's not related to Pharo.

**Comment by Ducasse** [comment]
Oh thanks for this information.

**Comment by jecisc** [comment]
We fixed something linked to SDL a few months ago that had some effects close to what is described in this issue. Can you retry?

**Comment by jordanmontt** [comment]
I tried and I couldn't reproduce it

**Comment by Rydier** [comment]
I cannot reproduce either anymore, closing!
